staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -165,6 +170,9 @@ type DelegationTarget interface {
 
 	// ListedPaths returns the paths for supporting an index
 	ListedPaths() []string
+
+	// OpenAPISpec returns the OpenAPI spec of the delegation target if exists, nil otherwise.
+	OpenAPISpec() *spec.Swagger
```

Comment:
This doesn't look right.  In the general case of openapi aggregation you don't have this level of access and given the unprotected handler chain from the delegate, this extra side-bar of communication shouldn't be necessary.